### PR TITLE
Ignore embedded scripts when asked

### PR DESCRIPTION
**Pull request by gjtorikian**
Closes https://github.com/gjtorikian/html-proofer/issues/233.

This PR proposes to add a new option, `ignore_script_embeds`, to resolve errors that read `Element script embeds close tag`.

I'm not 100% sure this is still the best way to handle this, but it might be the only way. From what I read in [Nokogiri's docs](http://www.rubydoc.info/github/sparklemotion/nokogiri/Nokogiri/XML/SyntaxError), there's pretty much no information given from a parse error.

/cc @akshayrawat

### CODE REVIEW

File: lib/html/proofer/checks/html.rb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # encoding: utf-8
 
 class HtmlCheck < ::HTML::Proofer::CheckRunner
+  class HtmlCheckable < ::HTML::Proofer::Checkable
+
+  end
```
Comment (penibelst) [inline lib/html/proofer/checks/html.rb]: Why is this class `HtmlCheckable` here?

Comment (gjtorikian) [inline lib/html/proofer/checks/html.rb]: Oh, I added it while testing something out. It's unnecessary.

**Comment by akshayrawat** [comment]
It would be nice if html-proofer supported multiple html validators like [vnu](https://github.com/validator/validator). My current setup disables html validation via proofer and instead uses vnu for that.

**Comment by gjtorikian** [comment]
> It would be nice if html-proofer supported multiple html validators like vnu. My current setup disables html validation via proofer and instead uses vnu for that.

That's a pretty good idea. It might be a pain to integrate a Java  executable into this Ruby library though.

File: spec/html/proofer/fixtures/html/ignore_script_embeds.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<script type="text/html" id="navbar-logged-in" data-proofer-ignore>
```
Comment (penibelst) [inline spec/html/proofer/fixtures/html/ignore_script_embeds.html]: I would remove `data-proofer-ignore` from this fixture. The spec “ignores embeded scripts when asked” must pass without the attribute.

**Comment by geobrando** [comment]
Hey @gjtorikian.  Was wondering if there's anything currently in the way of merging this.

**Comment by gjtorikian** [comment]
@geobrando Sorry! This is now out in 2.4.0: https://github.com/gjtorikian/html-proofer/releases/tag/v2.4.0

**Comment by geobrando** [comment]
Thanks!

**Comment by geobrando** [comment]
@gjtorikian 

I'm wondering if I'm missing something.  Either I don't know what I'm doing or there is something wrong with the tests.
```
  ruby:
    interpreter:  "ruby"
    version:      "2.1.2p95"
    date:         "2014-05-08"
    platform:     "i686-linux"
    patchlevel:   "2014-05-08 revision 45877"
    full_version: "ruby 2.1.2p95 (2014-05-08 revision 45877) [i686-linux]"
```
---

```ruby
require 'bundler/inline'

gemfile(true) do
  source 'https://rubygems.org'
  gem 'html-proofer', github: 'gjtorikian/html-proofer', :tag => 'v2.4.2'
end    
  
HTML::Proofer.new("./files",{:check_html => true, :ignore_script_embeds => true}).run
```
running with the fixture in `/files` yields

```
- ./files/repo-test.html
  *  Element script embeds close tag (line 2)
  *  Element script embeds close tag (line 2)
/home/geobrando/.rvm/gems/ruby-2.1.2/bundler/gems/html-proofer-f1f645f19a87/lib/html/proofer.rb:190:in `print_failed_tests': HTML-Proofer found 2 failures! (RuntimeError)
	from /home/geobrando/.rvm/gems/ruby-2.1.2/bundler/gems/html-proofer-f1f645f19a87/lib/html/proofer.rb:92:in `run'
	from /home/geobrando/Projects/htmlproofer-testing/test.rb:8:in `<main>'
```

**Comment by geobrando** [comment]
Hey @gjtorikian. Is this working as intended?

**Comment by colemanm** [comment]
I'm also still seeing the `Element script embeds close tag` errors when I have this new option enabled, running 2.5.1 latest. Same as @geobrando with `check_html` and `ignore_script_embeds` both true.

**Comment by gjtorikian** [comment]
The documentation is wrong, but the feature is working. The options should be passed like this:

``` ruby
opts = { :check_html => true, :validation => { :ignore_script_embeds => true } }
```

That is, within the `:validation` namespace (because I anticipated future options).

**Comment by geobrando** [comment]
@gjtorikian Ah.  You are correct.  Looking at the [binary](https://github.com/gjtorikian/html-proofer/blob/1aef3dcd9c1fec53b31bde99b7c63bc7c11473c9/bin/htmlproof), I think something needs to be changed to produce the proper namespaced option when using `--ignore-script-embeds` on  the CL but I couldn't figure out how to do it.  

From my limited knowledge of ruby it looks like you're mapping the CL flags directly to options that you pass to `HTML:proofer`.  I'll let you figure out how to do this as I assume you'd have to take into account adding additional options in the `:validation` namespace.

**Pull request merged**